src/ReefGuideAPI.jl

```diff
@@ -55,79 +55,93 @@ Load regional data to act as an in-memory cache.
-                parq_file = replace(first(g), ".tif" => "_lookup.parq")
-
-                if occursin("slope", string(dp))
-                    slope_table = GeoParquet.read(parq_file)
-                elseif occursin("flat", string(dp))
-                    flat_table = GeoParquet.read(parq_file)
-                else
-                    msg = "Unknown lookup found: $(parq_file). Must be 'slope' or 'flat'"
-                    throw(ArgumentError(msg))
+                push!(data_paths, first(g))
+                push!(data_names, string(k))
+                if occursin("valid", string(dp))
+                    # Load up Parquet files
+                    parq_file = replace(first(g), ".tif" => "_lookup.parq")
+
+                    if occursin("slope", string(dp))
+                        slope_table = GeoParquet.read(parq_file)
+                    elseif occursin("flat", string(dp))
+                        flat_table = GeoParquet.read(parq_file)
+                    else
+                        msg = "Unknown lookup found: $(parq_file). Must be 'slope' or 'flat'"
+                        throw(ArgumentError(msg))
+                    end
                 end
             end
+
+            # Pre-extract long/lat coordinates
+            coords = GI.coordinates.(slope_table.geometry)
+            slope_table[!, :lons] .= first.(coords)
+            slope_table[!, :lats] .= last.(coords)
+
+            coords = GI.coordinates.(flat_table.geometry)
+            flat_table[!, :lons] .= first.(coords)
+            flat_table[!, :lats] .= last.(coords)
+
+            rst_stack = RasterStack(data_paths; name=data_names, lazy=true)
+            regional_assessment_data[reg] = RegionalCriteria(
+                rst_stack,
+                slope_table,
+                flat_table
+            )
         end
 
-        # Pre-extract long/lat coordinates
-        coords = GI.coordinates.(slope_table.geometry)
-        slope_table[!, :lons] .= first.(coords)
-        slope_table[!, :lats] .= last.(coords)
-
-        coords = GI.coordinates.(flat_table.geometry)
-        flat_table[!, :lons] .= first.(coords)
-        flat_table[!, :lats] .= last.(coords)
-
-        rst_stack = RasterStack(data_paths; name=data_names, lazy=true)
-        regional_assessment_data[reg] = RegionalCriteria(
-            rst_stack,
-            slope_table,
-            flat_table
-        )
+        # Store cache on disk to avoid excessive cold startup times
+        @debug "Saving regional data cache to disk"
+        serialize(reg_cache_fn, regional_assessment_data)
+
+        # Remember, `@eval` runs in global scope.
+        @eval const REGIONAL_DATA = $(regional_assessment_data)
     end
 
-    # Store cache on disk to avoid excessive cold startup times
-    @debug "Saving regional data cache to disk"
-    serialize(reg_cache_fn, regional_assessment_data)
+    reef_outline_path = joinpath(reef_data_path, "rrap_canonical_outlines.gpkg")
```

Comment:
isn't this in the wrong spot? should be above `@debug "Saving regional data cache to disk"`
this function is getting a bit long, probably better to move everything in the `        @debug "Setting up regional data store..."
` else block to a function.